test/gtest/uct/test_sockaddr.cc

```diff
@@ -1043,6 +1059,95 @@ UCS_TEST_P(test_uct_sockaddr, ep_disconnect_err_codes)
 UCT_INSTANTIATE_SOCKADDR_TEST_CASE(test_uct_sockaddr)
 
 
+class test_uct_sockaddr_listener : public test_uct_sockaddr
+{
+protected:
+    test_uct_sockaddr_listener()
+    {
+        m_close_listener_notify = false;
+        m_server_uct_listener   = NULL;
+    }
+
+    static void conn_request_in_parallel_with_listener_close_cb(
+            uct_listener_h listener, void *arg,
+            const uct_cm_listener_conn_request_args_t *conn_req_args)
+    {
+        test_uct_sockaddr_listener *self =
+                reinterpret_cast<test_uct_sockaddr_listener *>(arg);
+
+        self->server_handle_conn_request(
+                listener, arg, conn_req_args,
+                test_uct_sockaddr::server_connect_cb,
+                test_uct_sockaddr::server_disconnect_cb);
+
+        ASSERT_TRUE(m_server_uct_listener == NULL);
+
+        // Notify listener_close_thread to destroy the listener
+        m_close_listener_notify = true;
+
+        // Wait until the listener is set from listener_close_thread and check
+        // whether the server's listener is set or not
+        self->wait_for_flag(&m_server_uct_listener);
```

Comment:
Consider making `m_server_uct_listener` an atomic since formally it is a race condition.